lua/entities/gmod_wire_expression2/core/quaternion.lua

```diff
@@ -527,33 +544,51 @@ end
 __e2setcost(13)
 
 --- Performs spherical linear interpolation between <q0> and <q1>. Returns <q0> for <t>=0, <q1> for <t>=1
+--- Derived from c++ source on https://en.wikipedia.org/wiki/Slerp
 e2function quaternion slerp(quaternion q0, quaternion q1, number t)
-	local dot = q0[1]*q1[1] + q0[2]*q1[2] + q0[3]*q1[3] + q0[4]*q1[4]
-	local q11
-	if dot<0 then
-		q11 = {-q1[1], -q1[2], -q1[3], -q1[4]}
-	else
-		q11 = { q1[1], q1[2], q1[3], q1[4] }  -- dunno if just q11 = q1 works
+	local dot = qDot(q0, q1)
+
+	if dot < 0 then
+		q1 = {-q1[1], -q1[2], -q1[3], -q1[4]}
+		dot = -dot
+	end
+
+	if dot > 0.9995 then
```

Comment:
Can you explain that magic number?

Reply:
This is the 'really small theta' case where the transcendental functions approximate to linear so it's cheaper to do this. I can add the comment though.